devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy

```diff
@@ -0,0 +1,334 @@
+			break
+	}
+
+	createEvent(name: "thermostatMode", value: mode, data: [supportedThermostatModes: state.supportedModes])
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatsetpointv2.ThermostatSetpointReport cmd) {
+	createEvent(name: "heatingSetpoint", value: convertTemperatureIfNeeded(cmd.scaledValue, 'C', cmd.precision), unit: temperatureScale)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd, sourceEndPoint = null) {
+	def map = [name: "temperature", value: convertTemperatureIfNeeded(cmd.scaledSensorValue, 'C', cmd.precision), unit: temperatureScale]
+	sendEventToChild(map)
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	if(cmd.parameterNumber == 3 && cmd.scaledConfigurationValue == 1) {
+		if(!childDevices) {
+			addChild()
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.applicationstatusv1.ApplicationBusy cmd) {
+	log.warn "Device is busy, delaying refresh"
+	runIn(15, "forcedRefresh", [overwrite: true])
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.warn "Unhandled command: ${cmd}"
+	[:]
+}
+
+def setThermostatMode(String mode) {
+	def modeValue = 0
+	switch (mode) {
+		case "heat":
+			modeValue = 1
+			break
+		case "emergency heat":
+			modeValue = 31
+			break
+		case "off":
+			modeValue = 0
+			break
+	}
+
+	[
+			secureEncap(zwave.thermostatModeV2.thermostatModeSet(mode: modeValue), 1),
+			"delay 2000",
+			secureEncap(zwave.thermostatModeV2.thermostatModeGet(), 1)
+	]
+}
+
+def heat() {
+	setThermostatMode("heat")
+}
+
+def off() {
+	setThermostatMode("off")
+}
+
+def emergencyHeat() {
+	setThermostatMode("emergency heat")
+}
+
+def setHeatingSetpoint(setpoint) {
+	setpoint = temperatureScale == 'C' ? setpoint : fahrenheitToCelsius(setpoint)
+	[
+			secureEncap(zwave.thermostatSetpointV2.thermostatSetpointSet([precision: 1, scale: 0, scaledValue: setpoint, setpointType: 1, size: 2]), 1),
+			"delay 2000",
+			secureEncap(zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: 1), 1)
+	]
+}
+
+def setThermostatSetpointUp() {
+	def setpoint = device.latestValue("heatingSetpoint")
+	if (setpoint < maxHeatingSetpointTemperature) {
+		setpoint = setpoint + (temperatureScale == 'C' ? 0.5 : 1)
+	}
+	setHeatingSetpoint(setpoint)
+}
+
+def setThermostatSetpointDown() {
+	def setpoint = device.latestValue("heatingSetpoint")
+	if (setpoint > minHeatingSetpointTemperature) {
+		setpoint = setpoint - (temperatureScale == 'C' ? 0.5 : 1)
+	}
+	setHeatingSetpoint(setpoint)
+}
+
+def refresh() {
+	def cmds = [
+			secureEncap(zwave.batteryV1.batteryGet(), 1),
+			secureEncap(zwave.batteryV1.batteryGet(), 2),
+			secureEncap(zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: 1)),
+			secureEncap(zwave.thermostatModeV2.thermostatModeGet()),
+			secureEncap(zwave.sensorMultilevelV5.sensorMultilevelGet()),
+			secureEncap(zwave.configurationV1.configurationGet(parameterNumber: 3))
```

Comment:
I'd recommend calling the configurationGet first so that we can create the child device ASAP. 

I had also added a call to `secureEncap(zwave.sensorMultilevelV5.sensorMultilevelGet(), 2)` from the refreshChild command.